### PR TITLE
Update index.md to add example of extending a property with an environment variable.

### PR DESCRIPTION
**Pull request by kmeilander**
Added example of extending a property with an environment variable.

### CODE REVIEW

**Review by sofietoft — COMMENTED** [review]
Thanks for updating the article @kmeilander !

I'll make sure it's reviewed as soon as possible 💪

**Comment by marcemarc** [comment]
Hi @kmeilander 

Thanks, It's a great addition to the example to let people know this is possible, and show how it's done.

Only thought, if people are coming to this cold, would it be worth clarifying where the example Environmental variable WEBSITE_DEPLOYMENT_ID is coming from (eg Azure webapp?).

regards

marc

**Comment by kmeilander** [comment]
I added a note about the variables being on the hosting server, and that the example is if using azure. 

Do you think that gives enough context?

File: Reference/Config/Serilog/index.md [hunk 1]
```diff
@@ -85,10 +85,12 @@ For example you could install the Nuget package `PM> Install-Package Serilog.Sin
...
 
-This is useful when you could be writing logs from all environments or multiple customer projects into a single logging source, such as Elasticsearch. This would allow you to search and filter for a specific project and its environment to see the log messages.
+This is useful when you could be writing logs from all environments or multiple customer projects into a single logging source, such as Elasticsearch. This would allow you to search and filter for a specific project and its environment to see the log messages.  You can also reference your hosting server's environment variables in the property values.
 
 In the `/config/serilog.user.config` file you can add the following lines, which the values could be changed or transformed as needed.
 
 ```xml
 <add key="serilog:enrich:with-property:customer" value="Super Customer" />
 <add key="serilog:enrich:with-property:environment" value="Production" />
+<add key="serilog:enrich:with-property:deploymentId" value="%WEBSITE_DEPLOYMENT_ID%" /> (if using Azure)
```
Comment (sofietoft) [inline Reference/Config/Serilog/index.md:95]: To keep the code snippet something you can easily implement in your own solution, I've added the mention of Azure in a comment instead, and also gave it a little more context.
What do you think? 🤔 

```suggestion
<add key="serilog:enrich:with-property:deploymentId" value="%WEBSITE_DEPLOYMENT_ID%" /> <!-- reference your hosting server's environment variables if using Azure -->
```

**Review by sofietoft — COMMENTED** [review]
Thanks for updating the article @kmeilander !

I've added a single suggestion - mind taking a look, and tell me what you think? :)

**Comment by sofietoft** [comment]
Hi again @kmeilander !

I've just had another look at this with the docs team.
The extra line you've added to the code snippet - does that require some extra package to be installed, by any chance? 🤔

**Comment by kmeilander** [comment]
I didn't think so but I can double check. 

It does require the environment var be set on the machine though.

**Comment by kmeilander** [comment]
It works out of the box for me (on 8.8.0)

you can throw this in a view file to see what env variable are available for you. 

```
<table>
    @foreach (var env in System.Environment.GetEnvironmentVariables().Keys)
    {
        <tr>
            <td>@env</td>
            <td>@(System.Environment.GetEnvironmentVariables()[env])</td>
        </tr>
    }
</table>

```

Also, I added it to serilog.config instead of the serilog.user.config file since I wanted that variable to be on all log entries. It looks like it does work if you add it in the serilog.user.config, but you need to also set up a separate writer to get your values.  This is true with the non environment values as well.    

I can't say I totally get what these 2 files are for and when to use each one, the documentation is a little confusing here.  I just tried one and then the other until I got what I wanted :)

**Comment by dampee** [comment]
I just tested this in 8.9.0 on my local machine with a new install, and I could not get this to work. What are the Serilog DLL's you are using?

**Comment by kmeilander** [comment]
It's working for me on 8.9.0 with a new install.  

What's your config value and where are you putting it?    Does it work with other non-environment variable custom properties? 

(Here is my value, I'm adding to the end of serilog.config)
`<add key="serilog:enrich:with-property:deploymentId" value="%MACHINENAME%" />`

And here is a log entry with the new custom property "deploymentId":

`{"@t":"2020-11-03T16:16:47.5738102Z","@mt":"Result for {HealthCheckName}: {HealthCheckResult}, Message: '{HealthCheckMessage}'","HealthCheckName":"Database data integrity check","HealthCheckResult":"Success","HealthCheckMessage":"<p>All media paths are valid</p>\r\n","SourceContext":"Umbraco.Web.HealthCheck.HealthCheckResults","ProcessId":35448,"ProcessName":"iisexpress","ThreadId":18,"AppDomainId":8,"AppDomainAppId":"LMW3SVC2ROOT","MachineName":"LAPTOP-041","Log4NetLevel":"INFO ","deploymentId":"LAPTOP-041"}`

**Comment by nul800sebastiaan** [comment]
Hi again @kmeilander! We're wrapping up the Hacktoberfest rewards list and haven't heard from you yet, make sure to fill in the form at https://docs.google.com/forms/d/e/1FAIpQLSdvfuqHFn7ib_xxMbVZ3nuyUgUTUCGPwmw0y83kiY4lP3OzKQ/viewform or if you want us to plant a tree on your behalf then you can reply as well and we'll take care of it. Thanks! 👍

**Comment by dampee** [comment]
Got it to work. I'll merge this!

**Comment by dampee** [comment]
Hi @kmeilander 

Sorry it took so long to merge this.  I've finally took some time to re-test this. I have tested this now against 8.11.1 and got it working.  So I am not sure why I did not got this working in the past.

Thanks a lot for your patience and your help!

**Pull request merged**